lib/util/stats.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const {
+  kConnected,
+  kPending,
+  kRunning,
+  kSize,
+  kFree,
+  kQueued
+} = require('../core/symbols')
+
+const kPool = Symbol('pool')
+const kClient = Symbol('client')
+
+class ClientStats {
+  constructor (client) {
+    this[kClient] = client
+  }
+
+  get connected () {
+    return this[kClient][kConnected]
+  }
+
+  get pending () {
+    return this[kClient][kPending]
+  }
+
+  get running () {
+    return this[kClient][kRunning]
+  }
+
+  get size () {
+    return this[kClient][kSize]
+  }
+}
+
+class PoolStats {
+  constructor (pool) {
+    this[kPool] = pool
+  }
+
+  get connected () {
+    return this[kPool][kConnected]
+  }
+
+  get free () {
+    return this[kPool][kFree]
+  }
+
+  get pending () {
+    return this[kPool][kPending]
+  }
+
+  get queued () {
+    return this[kPool][kQueued]
+  }
+
+  get running () {
+    return this[kPool][kRunning]
+  }
+
+  get size () {
+    return this[kPool][kSize]
+  }
+}
```

Comment:
I don't really understand this design. Instead of using getters, I would allocate new objects. This ensures that data is stable in time without having to copy them.

Reply:
To be fair I also didn't but this it the current implementation and I am following it assuming it has reasoning.

Reply:
👋🏼 What's the implicit agreement here? Refactor as a follow-up, not at all or here?

Reply:
@metcoder95 what's missing on this PR to land this? Anything expected from my side?

Reply:
Just applying the suggested refactor from getters to a plain object (it can also come from a class, but avoiding the getters/setters).

Reply:
Sure. From what I understand [55566bb](https://github.com/nodejs/undici/pull/4157/commits/55566bbf01230ca947e5a13a3c3da12302876400) is what you had in mind?